fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/registry/LandPathNodeTypesRegistry.java

```diff
@@ -0,0 +1,304 @@
+		}
+	}
+
+	/**
+	 * Gets the {@link PathNodeType} from the provider registered for the specified block at the specified position.
+	 *
+	 * <p>If no valid {@link PathNodeType} provider is registered for the block, it returns {@code null}.
+	 * You cannot use this method to retrieve vanilla block node types.
+	 *
+	 * @param state    Current block state.
+	 * @param world    Current world.
+	 * @param pos      Current position.
+	 * @param neighbor Specifies if the block is not a directly targeted block, but a neighbor block in the path.
+	 * @return the custom {@link PathNodeType} from the provider registered for the specified block,
+	 * passing the block state, the world, and the position to the provider, or {@code null} if no valid
+	 * provider is registered for the block.
+	 */
+	@Nullable
+	public static PathNodeType getPathNodeType(BlockState state, BlockView world, BlockPos pos, boolean neighbor) {
+		Objects.requireNonNull(state, "BlockState cannot be null!");
+		Objects.requireNonNull(world, "BlockView cannot be null!");
+		Objects.requireNonNull(pos, "BlockPos cannot be null!");
+
+		// Gets the node type provider for the block.
+		PathNodeTypeProvider provider = NODE_TYPES.get(state.getBlock());
+
+		//Gets the node type from the registered provider.
+		if (provider instanceof StaticPathNodeTypeProvider staticProvider) {
+			return staticProvider.getPathNodeType(state, neighbor);
+		} else if (provider instanceof DynamicPathNodeTypeProvider dynamicProvider) {
+			return dynamicProvider.getPathNodeType(state, world, pos, neighbor);
+		}
+
+		//If no valid provider is found returns null.
+		return null;
+	}
+
+	/**
+	 * Gets the {@link PathNodeType} from the provider registered for the specified block state.
+	 *
+	 * <p>This searches only for static providers.
+	 *
+	 * <p>If no valid {@link PathNodeType} provider is registered for the block, it returns {@code null}.
+	 * You cannot use this method to retrieve vanilla block node types.
+	 *
+	 * @param state    Current block state.
+	 * @param neighbor Specifies if the block is not a directly targeted block, but a neighbor block in the path.
+	 * @return the custom {@link PathNodeType} from the static provider registered for the specified block,
+	 * passing the specified block state to the provider, or {@code null} if no valid static provider
+	 * is registered for the block.
+	 */
+	@Nullable
+	public static PathNodeType getStaticPathNodeType(BlockState state, boolean neighbor) {
+		Objects.requireNonNull(state, "BlockState cannot be null!");
+
+		// Gets the node type provider for the block.
+		PathNodeTypeProvider provider = NODE_TYPES.get(state.getBlock());
+
+		//Gets the node type from the registered provider.
+		if (provider instanceof StaticPathNodeTypeProvider staticProvider) {
+			return staticProvider.getPathNodeType(state, neighbor);
+		}
+
+		//If no valid provider is found returns null.
+		return null;
+	}
+
+	/**
```

Comment:
All the methods below this exist mostly for Lithium, so I think they should be kept to a minimal API surface.
I wonder if we couldn't replace them all by a single `@Nullable PathNodeTypeProvider getProvider(Block block)` ? I'm not sure that calling such a method from outside the class would be allowed.

Reply:
I don't want to expose `PathNodeTypeProvider` to avoid further extensions that could break everything.  
I thought... maybe a better way is to add a single method `int getRegistrationType(Block block)` that returns 0 if the block is not registered, 1 if is registered static, 2 if is registered dynamic.  
These values may be enum constants.  
And yes i think only lithium will use this method.